Overhaul Slack participation and self-promotion guidelines (#1872)
writethedocs/www: docs/slack.rst

People who are new to our community in particular should be assisted and welcomed.

Our :doc:`/code-of-conduct` applies to all spaces associated with Write the Docs, which includes our Slack.
In addition, we have several Slack-specific guidelines:

Social tips
~~~~~~~~~~~

To have good experiences on Slack, create the conversations you want to be a part of.
You can help us do this by following these tips:

- **Say hi.**
  Don't forget to introduce yourself in ``#intros``.
  If you've been away a while, reintroduce yourself.
  The world is a better place with a flood of hello's.
- **Use inclusive language.**
  Our community is multicultural, multiracial, multigendered, and generally diverse.
  When posting in channels, avoid using words such as "guys" to address a mixed group; possible alternatives include "folks", "everyone", and "documentarians".
- **Remember that we're all documentarians.**
  The Write the Docs community is much more than a space for technical writers.

>   We welcome everyone who wants to learn more about creating documentation, including developers, testers, customer support agents, technical writers, program/product managers, developer relations specialists, and more.

Reviewer: Shall we add a "don't diss any of these groups" sentence here?

Author: I liked the suggestion to act on this further down. See https://github.com/writethedocs/www/pull/1872/files#r1067046560 and 65b5bc69

- **Only participate when you can be constructive.**
  Try to be constructive in all that you do, and otherwise consider not participating in a conversation.
- **Don't be snarky.**
  We are all trying the best we can.
- **Be careful with jokes.**
  You know that you're joking, but some people might take you seriously.
- **If you don't have anything nice to say, don't say anything at all.**

We also recommend reading `Microaggressions: Making space for everybody <https://www.metafilter.com/microaggressions.mefi>`_, which goes over how to avoid small slights and hurtful behaviors, especially against members of marginalized groups.

Online community issues
~~~~~~~~~~~~~~~~~~~~~~~

There is a specific set of behaviors that is prevalent in online communities that we'd like to address, to make sure our community doesn't fall into these traps. 
Please avoid these behaviors:

> - **No Black Sheep**. - There may be tools that you don't like.

Reviewer: Or here (re don't diss developers)....

Author: Yeah, this is a good place to mention it. Revised with 65b5bc69.

We expect everyone to follow these "rules of the road" at all times, to keep the conversation going smoothly and to reduce annoyances for everyone.

- **Start and stay on topic.**
  Digressions happen, but don't be afraid to move a conversation to a thread or another channel when it makes sense.
- **Focus on quality over quantity.**
  Excessive crossposting or reposting can drown out other conversations and looks like spam.
  Send your message to the most-relevant channel then give some time for conversations to start before casting a wider net.
  Sometimes it makes sense to send a similar message to multiple channels at once, or to post again, but show restraint.
- **Include context with links.**
  Give us your perspective: tell us what you're sharing, why, and your connection to it.
  Don't post a bare URL or use link shorteners: they're not informative and look like spam.
- **Use content warnings and threads for sensitive topics.**
  Write the Docs is a big community: topics that seem safe to you could make others feel uncomfortable or unwelcome.
  Help others choose which conversations they opt into by signposting sensitive or traumatic topics and confining them to threads.

> Posts not meeting these criteria may be deleted.

Reviewer 2: ```suggestion
Posts not meeting these criteria may be deleted, generally after a warning from our :ref:`moderation team <moderation-team>`.
```

This seems a bit abrupt and a bit self-evident -- could be good to soften it.

Author: Good idea. And this might be a good place to link to the moderation action section—linking actions to consequences. I'll make an edit for this.

Author: Done with. 64d01d6b

Self-promotion, surveys, and sales
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> You're welcome to share, link to, and mention your own work, including blog posts, social media, events, and paid products or services, but posts where you stand to benefit from others' money, personal information, or labor, come with higher expectations.

Reviewer: ```suggestion
You're welcome to share, link to, and mention your own work, including blog posts, social media, events, and paid products or services. That said, posts where you stand to benefit from others' money, personal information, or labor, come with higher expectations.
```

When this applies, make sure you follow the general guidelines closely.

In addition to the guidance that applies to all participants, we expect self-promotion on the Write the Docs Slack to follow these additional guidelines:

- **Post as yourself, not as a brand.**

>   Write the Docs Slack accounts must be operated by an individual human being that exists outside of work.

Reviewer 2: This feels like something larger than this line item, and probably worth mentioning closer to the top and without the caveat of being self-promotional? 

Not a blocker, but I think it's a really important point and something we implicitly understand, but we likely don't actually say. This is a great start -- actually saying it!

Reminds me of the long-standing tradition of not printing employer names on conference badges, and probably worth explaining our thinking behind this more explicitly in general. 🤔

Author: You're right, this is a more general rule. I'll move the whole bullet to the posting rules section.

It might be nice to include some similar text in the signup form, actually, since that's more or less _when_ people are most likely going to "violate" the rule. It might be a good addition to the email field (e.g., "Don't use a shared inbox or alias. Write the Docs Slack accounts are for real people, not brands. An account must be operated by an individual human being.")

Author: Moved the bullet with bb77c987.

Reviewer 2: Love the idea of adding it to the signup form. I actually don't know if we have control over that though..

- **Be transparent.**
  Clearly explain what you are promoting, why it matters to documentarians, and your involvement.
  Be up front about paywalls, registration requirements, referral benefits, or commissions.
  If you're running a survey, say when and how the results will be published.
- **Give more than you take.**
  Make self-promotion a reasonable proportion of your participation in the community.
  Too much self-promotion creates distrust and hostility.
  If you're not sure if it's too much, aim to help others 10 times for every time you self-promote.
- **Post publicly instead of sending direct messages (DMs)**.
  A DM is like a home and you're like a vampire: you're not allowed in without an invitation.
  It can be tempting to DM people, but please respect their inboxes and post in public channels unless invited to DM.
- **Read the room.**
  The Write the Docs Slack is informal and conversational; high-pressure sales aren't welcome.
  If people aren't interested in what you're offering, don't take it as a license to become louder and more aggressive.
  If you're asked to stop, then stop.

Here are some channels specifically for wholesome self-promotion:

> - ``#job-posts-only`` and ``#freelance`` are the best places to paid jobs

Reviewer: ```suggestion
- ``#job-posts-only`` and ``#freelance`` are the best places to post paid jobs and freelance work opportunities
```

> - ``#job-posts-only`` and ``#freelance`` are the best places to post paid jobs and freelance work opportunities

Reviewer 2: Might be worth splitting these into 2 line items, since people get annoyed when they are mixed up?

Author: Good catch. Fixed with a40111f9.

- ``#community-help-wanted`` is a good place to share calls for participation on open-source projects
- ``#???`` is a good place to announce non-Write the Docs events

> - ``#???`` is a good place to share a survey or calls for research interview subjects

Author: We discussed establishing channels for these specifically. Happy to update the PR, if we want to settle on some names?

Ideas:

- Rename `#calls-for-papers` to `#community-events`?
- Create `#community-research-help-wanted`? (or change the description/rules for `#community-help-wanted`?)

Reviewer: Did my comment here get lost? tl;dr

* yes to the first (broad for events not just cfps)
* no to the second (keeping it broad for various requests not just research)

Author: Right now, `#community-help-wanted` says it's for open source only. So I guess there's three options there:

1. Open a separate channel for surveys and interviews (i.e., `#community-help-wanted` continues to be open source only)
1. Permit non-open source in `#community-help-wanted` (i.e., change the channel description)
1. Do nothing (i.e., don't bless anything specific for surveys and interviews) except delete the line from the PR

Reviewer 3: I think I like option #2 - I don't know if people would be upset about losing the focus, but I think a broader spectrum for requests is okay.

Overall, these changes look fabulous - thank you so much for putting so much time into it, @ddbeck! 💥

Reviewer: My vote for option 2

> - ``#community-showcase`` is a space for shame-free self-promotion

Reviewer: "shame free" feels .... something. I would not want to it to be interpreted as without rules.

Author: This was much too subtle on my part. I wanted to make a little contrast here between shamelessness on the part of the self-promoter and shaming on the part of the audience. If folks are otherwise following the guidelines for self-promotion, then they shouldn't be made to feel bad for doing responsible self-promotion.

And now that I've written that out, maybe something like this would do?

```suggestion
- ``#community-showcase`` is a space for responsible self-promotion
```

> - ``#community-showcase`` is a space for responsible self-promotion

Reviewer 2: Do we care about trailing periods here? I think the rest of the lists in this doc have them, so might be good to keep them the same.

Author: Yes, we do care. 😂 Fixed with 9c3cf0ca.

> Reporting Issues

Reviewer 2: Much more explicit 💯

~~~~~~~~~~~~~~~~

> If you see a message in a public channel that looks like spam, you can react to it with the ``:spam:`` reactji to bring the moderators' attention to it.

Reviewer 2: ```suggestion
If you see a message in a public channel that looks like spam, you can react to it with the ``:spam:`` reactji to bring the moderators' attention to it.
Note that this is a public action, so only take it if you're okay with the user seeing you have done this. 
```

If you see a message in a public channel that looks like spam, you can react to it with the ``:spam:`` reactji to bring the moderators' attention to it.

If you see people not following the behavior guidelines you can:

> * Message them privately, and explain why what they said might have been hurtful to someone, referring to this guide.

Reviewer: This is an exception to "don't DM without invite"? hmmm

Author: Yeah, I was thinking about that too. Revised this a bit with 0e77b401.

Reviewer: It's tricky, because some folks might well respond better to a private "have you thought about..." instead of public "chastisment"...

Author: I agree with that. I think a few things come together here:

- We only explicitly advise against DMs for self-promotion. I think folks engaged in other kinds of conversations can and will use their judgement about taking it to DMs.
- For this particular case, I dropped the suggestion to DM. But I don't think we're forbidding it either.
- Sometimes a public chastisement is in the community's interest. Stating our values in public is a way to communicate to the victims of bad actors as well the bad actors themselves. But I'm skeptical we can train the community, in general, to handle those nuances in a document like this.

People who are new to our community in particular should be assisted and welcomed.

Our :doc:`/code-of-conduct` applies to all spaces associated with Write the Docs, which includes our Slack.
In addition, we have several Slack-specific guidelines:

Social tips
~~~~~~~~~~~

To have good experiences on Slack, create the conversations you want to be a part of.
You can help us do this by following these tips:

- **Say hi.**
  Don't forget to introduce yourself in ``#intros``.
  If you've been away a while, reintroduce yourself.
  The world is a better place with a flood of hello's.
- **Use inclusive language.**
  Our community is multicultural, multiracial, multigendered, and generally diverse.

>   When posting in channels, avoid using words such as "guys" to address a mixed group; possible alternatives include "folks", "everyone", and "documentarians".

Reviewer 4: ```suggestion
  When posting in channels, avoid using gendered words such as "guys" to address a mixed group; possible alternatives include "folks", "everyone", and "documentarians". 
  We know people have opinions on this - and we still want to encourage everyone to try shifting towards more inclusive global greetings.
```

Maybe too much? But I can see the "guys" thing raising some hackles.

Author: @ravindk89 I'm going to suggest leaving all three of these bullets as is, at least for the moment. A few reasons for this:

- They're some of the oldest bits of the doc. Since we haven't needed to change them in a long time, it suggests that they're either successful as is or at least harmless.
- To me, they're some of the least actionable pieces of the guidelines. While I think they set a particular tone, it's almost never that I find myself citing one of these bullets to someone as part of a moderation action.
- Given the two preceding points, I wonder if we might be better served by fewer words (e.g., reducing them to the bolded bits). I think these bullets establish the spirit of our guidelines (don't be a jerk), while the remainder outlines specific behaviors we'll actually attempt to moderate.

Reviewer 4: No objections - still LGTM!

- **Remember that we're all documentarians.**
  The Write the Docs community is much more than a space for technical writers.
  We welcome everyone who wants to learn more about creating documentation, including developers, testers, customer support agents, technical writers, program/product managers, developer relations specialists, and more.
- **Only participate when you can be constructive.**
  Try to be constructive in all that you do, and otherwise consider not participating in a conversation.
- **Don't be snarky.**
  We are all trying the best we can.
- **Be careful with jokes.**

>   You know that you're joking, but some people might take you seriously.

Reviewer 4: ```suggestion
  You know that you're joking, but some people might take you seriously.
  This is not a "no one can make jokes anymore" kinda rule.
  It's a "jokes that punch down aren't jokes" kinda rule.
```

Another "maybe too much, but also might head off some hackles. heckles? hackles.

> - **If you don't have anything nice to say, don't say anything at all.**

Reviewer 4: ```suggestion
- **If you don't have anything nice to say, don't say anything at all.**
This is complementary to the 'be constructive' rule. Again, not a "no one can be critical of anything" kinda rule, it's a "You can't be rude and pretend you're being helpful" rule.
```

Here are some channels specifically for wholesome self-promotion:

- ``#job-posts-only`` and ``#freelance`` are the best places to post paid jobs and freelance work opportunities

> - ``#community-events`` (formerly known as `#calls-for-papers`) is a good place to announce non-Write the Docs events

Reviewer: Let's make sure we update the channel description when we do this to mention the previously known as
